File: backend/app/models/rating.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime
from sqlalchemy.sql import func
from app.database.base import Base

class Rating(Base):

    __tablename__ = "ratings"

    id = Column(Integer, primary_key=True)

    user_id = Column(Integer, ForeignKey("users.id"))

    movie_id = Column(Integer, ForeignKey("movies.id"))

    rating = Column(Integer)

    created_at = Column(DateTime, server_default=func.now())

File: frontend/features/ratings/services/rating.ts
import api from "@/services/api";

export const RatingService = {

    createRating(
        userId: number,
        movieId: number,
        rating: number
    ) {
        return api.post(
            "/ratings/",
            {
                user_id: userId,
                movie_id: movieId,
                rating
            }
        );
    }
};